### PR TITLE
Refactor REST field names

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

Following the [internal documentation](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit) some fields name need to be refactored

### CODE REVIEW

**Review by tmetzke — COMMENTED** [review]
👍 Great stuff, thanks!
🔧 We need to adjust some tests as well that feature fields in JSON strings with the old names.
❓ Can we also rename `decisionKey` to `decisionDefinitionKey`? If I am not mistaken and based on discussions in internal channels, that should be more correct.

**Comment by nicpuppa** [comment]
@tmetzke I'd suggest to review the changes commit by commit

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -1384,7 +1384,7 @@ paths:
   /decisions/evaluation:
```
[the remark this answers is not in the document]
Comment (tmetzke) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1384]: 🔧 We also have to change the controller path for this one.

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -3818,21 +3818,21 @@ components:
...
           type: string
           description: The name of the DMN decision.
-        decisionVersion:
+        decisionDefinitionVersion:
           type: integer
           format: int32
           description: The version of the decision.
-        decisionType:
+        decisionDefinitionType:
           $ref: '#/components/schemas/DecisionInstanceTypeEnum'
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3836]: Is this a decision INSTANCE type or decision DEFINITION type? @ralfpuchert, any input here?

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 3]
```diff
@@ -4008,7 +4008,7 @@ components:
       properties:
         process:
           $ref: "#/components/schemas/DeploymentProcess"
-        decision:
+        decisionDefinition:
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:4011]: 🔧 If we go for `decisionDefinition` here, we should go for `processDefinition` above as well, I think.

**Review by tmetzke — COMMENTED** [review]
🚀 What a PR!!!
👍 Looks really good already.
🔧 I found a few minor things to look at but overall this looks fantastic 🦖

File: clients/java/src/test/java/io/camunda/zeebe/client/decision/rest/DecisionEvaluationRestTest.java [hunk 4]
```diff
@@ -62,26 +62,26 @@ public class DecisionEvaluationRestTest extends ClientRestTest {
...
-              .decisionKey(DECISION_KEY)
-              .decisionId("my-decision")
-              .decisionName("My Decision")
-              .decisionVersion(1)
-              .decisionOutput("testOutput")
+              .decisionDefinitionKey(DECISION_KEY)
+              .decisionDefinitionId("my-decision")
+              .decisionDefinitionName("My Decision")
+              .decisionDefinitionVersion(1)
+              .decisionDefinitionOutput("testOutput")
```
[the remark this answers is not in the document]
Comment (nicpuppa) [reply clients/java/src/test/java/io/camunda/zeebe/client/decision/rest/DecisionEvaluationRestTest.java:81]: Waiting for input here (I'd go for `output` probably)

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml (hunk 2 above)
Comment (nicpuppa) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3836]: Waiting for inputs

File: clients/java/src/test/java/io/camunda/zeebe/client/decision/rest/DecisionEvaluationRestTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (tmetzke) [reply clients/java/src/test/java/io/camunda/zeebe/client/decision/rest/DecisionEvaluationRestTest.java:81]: Yeah, let's go for `output` 👍

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 5]
```diff
@@ -3593,16 +3593,16 @@ components:
     EvaluateDecisionRequest:
       type: object
       properties:
-        decisionKey:
+        decisionDefinitionKey:
           description: |
             The unique key identifying the decision to be evaluated.
             Cannot be used together with decisionId.
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3599]:
```suggestion
            Cannot be used together with decisionDefinitionId.
```

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml (hunk 2 above)
Comment (houssain-barouni) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3836]: yes this should be a `DecisionDefinitionType`

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 6]
```diff
@@ -1381,10 +1381,10 @@ paths:
                 $ref: "#/components/schemas/ProblemDetail"
 
 
-  /decisions/evaluation:
+  /decision-definitions/evaluation:
```
[the remark this answers is not in the document]
Comment (nicpuppa) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1384]: Wdym ?

[the remark this answers is not in the document]
Comment (nicpuppa) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1384]: Nevermind

[the remark this answers is not in the document]
Comment (tmetzke) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1384]: 😄

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/DecisionController.java [hunk 7]
```diff
@@ -23,7 +23,7 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 
 @CamundaRestController
-@RequestMapping("/v2/decisions")
+@RequestMapping("/v2/decision-definitions")
 public class DecisionController {
```
Comment (houssain-barouni) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/DecisionController.java:27]:
```suggestion
public class DecisionDefinitionController {
```

**Review by tmetzke — APPROVED** [review]
This is awesome, thanks a lot, @nicpuppa 🚀

**Pull request merged**